pulp_ansible/app/tasks/collections.py

```diff
@@ -608,13 +626,61 @@ async def _find_all_collections(self):
 
         await asyncio.gather(*tasks)
 
+    async def _can_be_optimized(self):
+        """Check last synced metadata time."""
+        root, api_version = await self._get_root_api(self.remote.url)
+        if api_version == 3:
+            downloader = self.remote.get_downloader(
+                url=root, silence_errors_for_response_status_codes={404}
+            )
+            try:
+                metadata = parse_metadata(await downloader.run())
+            except FileNotFoundError:
+                return False
+
+            sources = set()
+            if self.collection_info:
+                sources = {r.source for r in self.collection_info if r.source}
+            sources.add(self.remote.url)
+
+            published = metadata.get("published")
+            repo_metadata_time = self.repository.last_synced_metadata_time
+            repo_metadata_time = repo_metadata_time.isoformat() if repo_metadata_time else ""
+
+            # From serializers.DateTimeField.to_representation:
+            if repo_metadata_time.endswith("+00:00"):
+                repo_metadata_time = repo_metadata_time[:-6] + "Z"
+
+            if published == repo_metadata_time and len(sources) < 2:
+                msg = _(
+                    "no-op: {repo} did not change since last sync - {published}".format(
+                        repo=self.remote.url, published=published
+                    )
+                )
+                noop = ProgressReport(message=msg, code="noop")
+                noop.state = TASK_STATES.COMPLETED
+                noop.save()
```

Comment:
It I think would be more reliable to report to the user that the no-op check occured anytime it did occur. To do that I would move this part higher up in the function before any `return` statements (maybe even at the beginning).

Reply:
this is what this code is doing, no-op only happens when return True, this is the only part when it returns True

Reply:
What I'm saying is the progress reporting should say that it is checking for a no-op regardless of it if occurs or not. In terms of reporting on what it's doing I expect this progress report to be there when optimize=True and not to be there when optimize=False.

Reply:
so how can I assert if no-op happened?
I added the ProgressReport just for it

Reply:
I expected it's not on asserting that the progress reporting shows no-op was checked or not, it's that the rest of the 'syncing, etc, etc' progress reports are not present and the state is `completed`.